src/main/java/roomescape/domain/reservation/api/ReservationCommandApi.java

```diff
@@ -0,0 +1,35 @@
+package roomescape.domain.reservation.api;
+
+import lombok.RequiredArgsConstructor;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+import roomescape.domain.reservation.api.response.ReserveHttpResponse;
+import roomescape.domain.reservation.dto.ReservationId;
+import roomescape.domain.reservation.service.ReservationCommandService;
+import roomescape.domain.reservation.service.request.ReserveRequest;
+import roomescape.domain.reservation.service.response.ReserveResponse;
+
+@RestController
+@RequiredArgsConstructor
+public class ReservationCommandApi {
+
+    private final ReservationCommandService commandService;
+
+    @PostMapping("/reservations")
+    public ResponseEntity<ReserveHttpResponse> reserve(
+            @RequestBody ReserveRequest request
+    ) {
+        request.validateAllFieldsExist();
+        final ReserveResponse response = commandService.reserve(request);
+        return ResponseEntity.ok().body(ReserveHttpResponse.from(response));
+    }
+
+    @DeleteMapping("/reservations/{reservationId}")
+    public ResponseEntity<Void> cancel(
+            @PathVariable(name = "reservationId") Long reservationId
```

Comment:
```suggestion
            @PathVariable Long reservationId
```

Gradle build 환경에서 패턴에서 사용한 변수명과 동일하다면 name을 별도로 지정하지 않아도 괜찮습니다.

> You can explicitly name URI variables (for example, @PathVariable("customId")), but you can leave that detail out if the names are the same and your code is compiled with the -parameters compiler flag.

- [Spring docs - URI Patterns](https://docs.spring.io/spring-framework/reference/web/webmvc/mvc-controller/ann-requestmapping.html#mvc-ann-requestmapping-uri-templates)

Reply:
찾아보니 [Spring Boot 3.2에서 파라미터 이름 인식 문제](https://www.inflearn.com/questions/1087879/pathvariable-name-%EC%83%9D%EB%9E%B5-%EC%A7%88%EB%AC%B8-%EB%93%9C%EB%A6%BD%EB%8B%88%EB%8B%A4)가 있더라고요. 
언급해주신대로 Gradle build 시에는 문제되지 않지만, 주로 IntelliJ IDEA 빌드를 사용하고 있는 관계로 name을 별도로 지정하게 되었습니다..!

---

조금 다른 얘기지만 브리 또한 제가 IntelliJ IDEA 빌드를 사용한 것으로 추측하고 Gradle build를 권해주신다는 느낌을 받았는데요, 브리는 IntelliJ IDEA 빌드 대신 Gradle build를 사용하신 이유가 있을까요? 저는 현재 속도가 더 빠르다는 이유로 IntelliJ IDEA를 사용하고 있긴 합니다 :)

Reply:
지금의 flag 이슈처럼 intellij 빌드가 gradle 빌드와 미묘하게 다른 부분들이 있는데요.
특정 pc에서만 빌드나 테스트가 실패하는데 아무리 코드를 뜯어봐도 문제점을 찾기 어려운 삽질을 몇 번 경험했어요 🥲
그 이후로는 local과 remote 빌드를 가능한 동일하게 할 수 있는 Gradle 빌드를 선호합니다. ㅋㅋ